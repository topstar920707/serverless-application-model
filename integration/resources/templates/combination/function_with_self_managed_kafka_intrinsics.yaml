Parameters:
  TopicsValue:
    Type: CommaDelimitedList
    Default: Topic

  KafkaBootstrapServersValue:
    Type: String
    Default: abc.xyz.com:9092

Resources:
  KafkaFunctionWithIntrinsics:
    Type: AWS::Serverless::Function
    Properties:
      Handler: index.handler
      Runtime: nodejs14.x
      CodeUri: ${codeuri}
      MemorySize: 128
      Events:
        MyKafkaClusterWithIntrinsics:
          Type: SelfManagedKafka
          Properties:
            KafkaBootstrapServers:
              - Ref: KafkaBootstrapServersValue
            Topics:
              Ref: TopicsValue
            SourceAccessConfigurations:
              - Type: BASIC_AUTH
                URI:
                  Ref: KafkaUserSecret

  KafkaUserSecret:
    Type: AWS::SecretsManager::Secret
    Properties:
      Name: KafkaUserPassword
      SecretString:
        Fn::Sub: '{"username":"testBrokerUserWithInstrinsic","password":"testBrokerPasswordWithInstrinsic"}'

