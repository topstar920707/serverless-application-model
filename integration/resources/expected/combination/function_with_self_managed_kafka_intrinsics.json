[
  {
    "LogicalResourceId": "KafkaFunctionWithIntrinsics",
    "ResourceType": "AWS::Lambda::Function"
  },
  {
    "LogicalResourceId": "KafkaFunctionWithIntrinsicsMyKafkaClusterWithIntrinsics",
    "ResourceType": "AWS::Lambda::EventSourceMapping"
  },
  {
    "LogicalResourceId": "KafkaFunctionWithIntrinsicsRole",
    "ResourceType": "AWS::IAM::Role"
  },
  {
    "LogicalResourceId": "KafkaUserSecret",
    "ResourceType": "AWS::SecretsManager::Secret"
  }
]